From 0383848e4e60d01ddcb39bb015b8fbea870a139a Mon Sep 17 00:00:00 2001
From: Hou Qi <qi.hou@nxp.com>
Date: Tue, 17 Oct 2023 14:26:22 +0800
Subject: [PATCH] LF-10264 v4l2: fix AV unsync after seek

v4l2_buffer sequence will be reset after flush, need to replace
it with v4l2_buffer timestamp.

upstream status: pending
Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/5479>
---
 sys/v4l2/gstv4l2allocator.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sys/v4l2/gstv4l2allocator.c b/sys/v4l2/gstv4l2allocator.c
index 81a489478..be72a1fa5 100644
--- a/sys/v4l2/gstv4l2allocator.c
+++ b/sys/v4l2/gstv4l2allocator.c
@@ -1429,7 +1429,7 @@ gst_v4l2_allocator_dqbuf (GstV4l2Allocator * allocator,
   if (V4L2_TYPE_IS_OUTPUT (obj->type)) {
     gst_v4l2_allocator_reset_size (allocator, group);
     if (group->buffer.flags & V4L2_BUF_FLAG_ERROR) {    //output buffer cannot be decoded
-      obj->drop_frames = g_list_append (obj->drop_frames, &group->buffer.sequence);
+      obj->drop_frames = g_list_append (obj->drop_frames, &group->buffer.timestamp);
       if (!obj->frame_decoded)
         obj->err_cnt++;
     } else {
-- 
2.17.0

